table/builder.go

```diff
@@ -219,17 +222,28 @@ func (b *Builder) Finish() []byte {
 		bf.Add(key)
 	}
 
+	// Write bloom filter.
+	b.tableIndex.BloomFilter = bf.JSONMarshal()
 	b.finishBlock() // This will never start a new block.
-	index := b.blockIndex()
-	b.buf.Write(index)
 
-	// Write bloom filter.
-	bdata := bf.JSONMarshal()
-	n, err := b.buf.Write(bdata)
+	index := b.blockIndex()
+	n, err := b.buf.Write(index)
 	y.Check(err)
+
+	// Write index size
 	var buf [4]byte
 	binary.BigEndian.PutUint32(buf[:], uint32(n))
 	b.buf.Write(buf[:])
 
+	// Build checksum for index
+	checksum, err := y.BuildChecksum(index)
+	y.Check(err)
+
+	// Write checksum to the file
+	n, err = b.buf.Write(checksum)
+
+	// Write len of the checksum to the file
+	binary.BigEndian.PutUint32(buf[:], uint32(n))
+	b.buf.Write(buf[:])
```

Comment:
return values are ignored? GolangCI is not complaining, is it correctly setup?

Reply:
No. Golang CI is not running. I'll figure out why.

Reply:
Also, fixed this in https://github.com/dgraph-io/badger/pull/816/commits/fc950be2fe51ca589fb19c77e33bf10cafb9c1bd